BH/Modules/Item/ItemDisplay.cpp

```diff
@@ -1150,236 +1455,504 @@ void Condition::BuildConditions(vector<Condition*>& conditions,
+		break;
+	case COND_WP2:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_MACE));
+		break;
+	case COND_WP3:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_SWORD));
+		break;
+	case COND_WP4:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_DAGGER));
+		break;
+	case COND_WP5:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_THROWING));
+		break;
+	case COND_WP6:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_THROWING | ITEM_GROUP_SPEAR));
+		break;
+	case COND_WP7:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_SPEAR));
+		break;
+	case COND_WP8:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_POLEARM));
+		break;
+	case COND_WP9:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_BOW));
+		break;
+	case COND_WP10:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_CROSSBOW));
+		break;
+	case COND_WP11:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_STAFF));
+		break;
+	case COND_WP12:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_WAND));
+		break;
+	case COND_WP13:
+		Condition::AddOperand(conditions, new ItemGroupCondition(ITEM_GROUP_SCEPTER));
+		break;
+	case COND_SK:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >(int)SKILL_MAX) { break; }
+		Condition::AddOperand(conditions, new ItemStatCondition(STAT_SINGLESKILL, cond_num, operation, value));
+		break;
+	case COND_OS:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >(int)SKILL_MAX) { break; }
+		Condition::AddOperand(conditions, new ItemStatCondition(STAT_NONCLASSSKILL, cond_num, operation, value));
+		break;
+	case COND_CHSK:
+		// skills granted by charges
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >(int)SKILL_MAX) { break; }
+		Condition::AddOperand(conditions, new ChargedCondition(operation, cond_num, value));
+		break;
+	case COND_CLSK:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >= CLASS_NA) { break; }
+		Condition::AddOperand(conditions, new ItemStatCondition(STAT_CLASSSKILLS, cond_num, operation, value));
+		break;
+	case COND_ALLSK:
+		Condition::AddOperand(conditions, new ItemStatCondition(STAT_ALLSKILLS, 0, operation, value));
+		break;
+	case COND_TABSK:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num > SKILLTAB_MAX) { break; }
+		Condition::AddOperand(conditions, new ItemStatCondition(STAT_SKILLTAB, cond_num, operation, value));
+		break;
+	case COND_STAT:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >(int)STAT_MAX) { return; }
+		Condition::AddOperand(conditions, new ItemStatCondition(cond_num, 0, operation, value));
+		break;
+	case COND_CHARSTAT:
+		if ((number_ss >> cond_num).fail() || cond_num < 0 || cond_num >(int)STAT_MAX) { return; }

```

Comment:
Is there a functional difference between `break` and `return` for these? (may have just been missed when you were switching things over)

Reply:
In this case, the difference is that the final line is run when it wasn't before:
`for (vector<Condition*>::iterator it = endConditions.begin(); it != endConditions.end(); it++) { Condition::AddNonOperand(conditions, (*it)); }`
I didn't understand why this was skipped before the conversion and only those specific conditions returned like this.

Reply:
cool cool. I noticed that few were swapped over from return to break so I just wanted to make sure it was intentional.